pkg/controllers/controller.go

```diff
@@ -95,6 +95,14 @@ func NewController(conf Config) (*Controller, error) {
 		nodeSelectorFunc = func(node *corev1.Node) bool {
 			return selector.Matches(labels.Set(node.Annotations))
 		}
+	} else if conf.ManageNodesWithLabelSelector != "" {
+		selector, err := labels.Parse(conf.ManageNodesWithLabelSelector)
+		if err != nil {
+			return nil, err
+		}
+		nodeSelectorFunc = func(node *corev1.Node) bool {
+			return selector.Matches(labels.Set(node.Labels))
+		}
```

Comment:
```suggestion
    // client-go supports label filtering, so ignore this.
	// } else if conf.ManageNodesWithLabelSelector != "" {
```

https://github.com/kubernetes-sigs/kwok/blob/a830f47bfb036704093ed06b5caf79c3c1723d5b/pkg/controllers/node_controller.go#L135

`client-go` supports a label selector, so this is not necessary, but we can add a comment to clarify it